### PR TITLE
[CELEBORN-634][LICENSE] Update LICENSE and NOTICE

### PR DESCRIPTION
**Pull request by zhongqiangczq**
### What changes were proposed in this pull request?

Update LICENSE and NOTICE to meet the requirements of ASF.

### Why are the changes needed?

Since the last release, some dependencies' version have been updated. 

- Flink is supported from flink-1.14 to flink-1.17
- Ratis's version has updated from 2.4.1 to 2.5.1 .
- Netty's version has updated from 4.1.77.Final to 4.1.92.Final

so NOTICE and LICENSE should be updated corresponding to that modification.

### Does this PR introduce _any_ user-facing change?

No, it's for legal propose, does not affect functionality.

### How was this patch tested?

Manuelly review.

### CODE REVIEW

File: NOTICE [hunk 1]
```diff
@@ -8,3 +8,5 @@ The Apache Software Foundation (https://www.apache.org/).
 Apache Spark
 Copyright 2014 and onwards The Apache Software Foundation
 
+Apache Flink
+Copyright 2014-2022 The Apache Software Foundation
```
Comment (pan3793) [inline NOTICE:12]: nit: and extra line at the end

File: LICENSE-binary [hunk 2]
```diff
@@ -241,6 +241,7 @@ io.netty:netty-transport-native-unix-common
 io.netty:netty-transport-rxtx
 io.netty:netty-transport-sctp
 io.netty:netty-transport-udt
+io.netty:netty-handler-ssl-ocsp
```
Comment (pan3793) [inline LICENSE-binary:244]: do we need the ssl? @FMX

**Comment by pan3793** [comment]
The PR title is corrupt, and please fill the PR description

File: LICENSE-binary (hunk 2 above)
Comment (FMX) [reply LICENSE-binary:244]: We do not need ssl now but the handle "ocsp" was introduced by Netty 4.1.86. I think we could exclude this dependency or accept this change. @pan3793

File: NOTICE (hunk 1 above)
Comment (zhongqiangczq) [reply NOTICE]: new line has added

File: LICENSE-binary (hunk 2 above)
Comment (pan3793) [reply LICENSE-binary:244]: I don't think the ssl work w/o other ssl jars, let's exclude it.

Comment (zhongqiangczq) [reply LICENSE-binary:244]: ok, i will remove netty-handler-ssl-ocsp.

**Review by pan3793 — APPROVED** [review]

**Comment by pan3793** [comment]
@zhongqiangczq after PR merged, please

- resolves the JIRA ticket correctly, specifically, fill the fixed version
- backport PR to release branch if needed
- left comments on which branches include this PR, e.g. `Merged to main/0.3`

**Pull request merged**